### PR TITLE
Add csv output option and domain selection to deku-benchmark 

### PR DESCRIPTION
**Pull request by InfiniteSwerve**
## Problem

We want to be able to write benchmark data to csv files, and choose the amount of domains from the command line
## Solution

This PR uses cmdliner to allow optional writing to a csv file, with a mandatory domain choice

### CODE REVIEW

File: output.csv [hunk 1]
```diff
@@ -0,0 +1,70 @@
+produce, 1000000, 0.489075
```
Comment (d4hines) [inline output.csv:1]: Please delete this file.

File: deku-p/src/core/bin/benchmark/deku_benchmark.ml [hunk 2]
```diff
@@ -267,15 +267,47 @@ let benches : (module BENCH) list =
...
+    Eio.Path.save ~append:true ~create:(`If_missing 0o600) (dir / file) data
+  in
+  Eio_main.run @@ fun env -> run (Eio.Stdenv.cwd env)
+
+let write_to_terminal ~data ~file =
+  let _ = file in
+  Format.eprintf "%s%!" data
+
+type params = {
+  domains : int; [@env "DEKU_DOMAINS"]
```
Comment (d4hines) [inline deku-p/src/core/bin/benchmark/deku_benchmark.ml:288]:
```suggestion
  domains : int; [@env "DEKU_DOMAINS"][@default 16]
```

Comment (d4hines) [reply deku-p/src/core/bin/benchmark/deku_benchmark.ml:288]: Also please delete the code starting on line 8 that defines `domains`.

**Review by d4hines — CHANGES_REQUESTED** [review]
See comments. Addiotionally can you:
- print a header row, something like: `benchmark, items, duration`. You might want to delete the file if it exists already.
- make the `items` column a number only - no text.

File: deku-p/src/core/bin/benchmark/deku_benchmark.ml (hunk 2 above)
Comment (InfiniteSwerve) [reply deku-p/src/core/bin/benchmark/deku_benchmark.ml]: Done, good catches.

**Comment by InfiniteSwerve** [comment]
@d4hines Branch has been rebased and I've modified the code based on your comments. If that's all and there's nothing else blocking, then please approve and merge.

**Comment by InfiniteSwerve** [comment]
@d4hines I've rebased and addressed your comments with the latest commit. The changes made to the writer function work, but I'm not sure whether I've made things too complex. If there are no other blocking issues then please merge this PR.

**Review by d4hines — APPROVED** [review]
:+1: 
[all might happy with progress](https://media.giphy.com/media/UJv4aOufZ3INq/giphy.gif)

**Pull request merged**